presto-hive/src/main/java/io/prestosql/plugin/hive/BackgroundHiveSplitLoader.java

```diff
@@ -41,6 +41,8 @@
 import org.apache.hadoop.fs.FileSystem;
```

Comment:
I have mentioned this in the doc https://docs.google.com/document/d/1GwSNc_6jSP8N1SyJIHnaMeWvh7JzhGPZ7QKm1DHJqEw

It is mainly to support multiple queries in one presto transaction.

Reply:
but shouldn't single presto transaction be mapped to single hive transaction? How would you implement `abort` then? Would you `abort` several transactions?

Reply:
Under one presto transaction only one query would be active at a time so only one hive transaction will be active under a presto transaction at any point of time. So abort will only abort the hive transaction for the currently running query.

Reply:
Correct. However you can run several queries sequentially within one Presto transaction. Like (in Presto):
```
BEGIN TRANSACTION;
INSERT...
INSERT...
UPDATE...
DELETE...
ROLLBACK;
```
is that possible in Hive? If so we would need to support that in Presto as well.

Reply:
No, as noted in the doc, Hive only supports autocommit

Reply:
And so Presto should only support autocommit then?

Reply:
I am not sure which one to pick. I had that point in the doc too but didnt get any comments on it so went ahead with supporting multiple queries as that seems user friendly.

Reply:
> And so Presto should only support autocommit then?

That's irrelevant now as we're discussing reading only.
From reader's perspective what matters is transaction isolation. 

Currently, Hive declares it supports READ_UNCOMMITTED only.
https://github.com/prestosql/presto/blob/b747ecb656b010b514aa75f938c5efc425e6603f/presto-hive/src/main/java/io/prestosql/plugin/hive/HiveConnector.java#L182-L184

With Hive transactional tables, this becomes READ_COMMITTED or REPEATABLE_READ. Which one it is, depends on how long we keep committed transactions state.

- if we acquire this per query, we're READ_COMMITTED
- if we acquire this per Presto transaction, we're REPEATABLE_READ
   - side effect is that we're freezing resources (locks) longer on the Hive side
   - when Presto transaction is idle (between queries), it can be harder to us to send heartbeats

In my opinion, READ_COMMITTED should be sufficient, so it should be enough to acquire committed transactions state on per query basis. (Per query != per table scan, of course).

Nonetheless, whatever choice we're taking, we should think how this will extend to writing in the future.

Reply:
@electrum

Reply:
I don’t think this is enough for higher isolation levels. Is the set of partitions is guaranteed to be stable across requests?

Reply:
Not sure I get the question correctly @electrum, are you asking if partitions' data can change while we have read lock on it? That can happen, Insert Into is allowed when shared read lock exists on partition. But if we start taking locks and storing write ids (db snapshot) at Presto transaction level then even if partition data changes the snapshot remains the same, so higher level of isolations can be possible.

Reply:
> But if we start taking locks and storing write ids (db snapshot) at Presto transaction level 

That would also mean we need to heartbeat for the whole duration of a transaction, so in between the queries.